pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -419,7 +419,8 @@ type GitDirectoryGeneratorItem struct {
 }
 
 type GitFileGeneratorItem struct {
-	Path string `json:"path" protobuf:"bytes,1,name=path"`
+	Path         string                `json:"path" protobuf:"bytes,1,name=path"`
+	FileSelector *metav1.LabelSelector `json:"fileSelector,omitempty" protobuf:"bytes,2,name=fileSelector"`
```

Comment:
Maybe name it FileLabelSelector to be more clear it's a label selector?

Reply:
Done